components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1565,7 +1566,16 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
                                 authenticatorId = addAuthenticator(connection, tenantID,
                                         ApplicationConstants.LOCAL_IDP_NAME,
                                         lclAuthenticator.getName(),
-                                        lclAuthenticator.getDisplayName());
+                                        lclAuthenticator.getDisplayName(),
+                                        lclAuthenticator.getDefinedByType().toString());
+                            } else {
+                                /* On demand migration for already saved local authenticators to save definedBy and
+                                 authenticationType properties to the database.
+                                 Remove this else block, once on-demand migration is done.*/
+                                log.debug("Authenticator already exists. Updating the authenticator.The defined "
+                                        + "by type is setting to: " + lclAuthenticator.getDefinedByType().toString());
```

Comment:
Rather than having a debug log IMO, it's good if we can have a separate system log recognized to track data level changes that the system has performed moving the state from one state to another

Reply:
No need this on-demand migration.